### PR TITLE
Some optimizations: Calculate values in setup instead of in innner loop

### PR DESCRIPTION
**Pull request by htgoebel**
* Zero-base columns internally, which avoids some calculations for every line.
* Pre-calculate prefixes, which avoids calculating them for each IP address.
* Format log messages only when actually output

### CODE REVIEW

File: anonip.py [hunk 1]
```diff
@@ -80,22 +80,43 @@ def __init__(
...
         self.increment = increment
         self.delimiter = delimiter
         self.replace = replace
         self.skip_private = skip_private
 
+    @property
+    def ipv4mask(self):
+        return self._ipv4mask
+
+    @ipv4mask.setter
```
Comment (open-dynaMIX) [inline anonip.py]: What is the reason for using getters and setters here?

Couldn't we just have this in `__init__()`?:

```python
self.prefixes = {
    4: 32 - ipv4mask,
    6: 128 - ipv6mask
}
```

Comment (open-dynaMIX) [reply anonip.py]: Nevermind, I get it. Still I'd prefer having a dict `self.prefixes` instead of the list.

Comment (htgoebel) [reply anonip.py]: My intention was to gain speed, as I assume accessing elements using an index into an array is faster then fetching a hash first. But as https://wiki.python.org/moin/TimeComplexity as well as my own tests show: dicts are as fast as lists in our case (and maybe even a small tick faster).

(Thinking in C-code I can not understand this, but, well, I did not trace the CPython code in detail ;-)

So I will change the code to use dicts.

Comment (htgoebel) [reply anonip.py]: For the records, her is the code I used for testing
```
import timeit

ipv4mask = 11
ipv6mask = 83
prefixes_dict = {
    4: 32 - ipv4mask,
    6: 128 - ipv6mask
}
prefixes_list = [None, None, None, None,
                 32 - ipv4mask, None, 128 - ipv6mask]

def from_dict(): return prefixes_dict[4]

def from_list(): return prefixes_list[4]

timeit.timeit('from_dict()', globals=globals())
timeit.timeit('from_list()', globals=globals())


sum(timeit.timeit('from_dict()', globals=globals()) for i in range(1000))
sum(timeit.timeit('from_list()', globals=globals()) for i in range(1000))
```

**Comment by htgoebel** [comment]
@open-dynaMIX I updated the code to use dicts. Read to merge IMHO.

File: anonip.py (hunk 1 above)
Comment (open-dynaMIX) [reply anonip.py]: Yeah, dicts are hashtables in python and therefore very performant.

**Comment by open-dynaMIX** [comment]
Great, thanks a lot @htgoebel!

**Pull request merged**